docs/developer-experience/going-further.md

```diff
@@ -0,0 +1,184 @@
+
+There are other tools you can think of:
+
+* Linting checks for your [markdown](https://github.com/DavidAnson/markdownlint) files,
+* Linting checks for your [bash](https://www.shellcheck.net/) scripts,
+* Etc...
+
+Linting checks on files that are not *the source code* can ensure a common format with common rules for each developer. These checks should be also run in a [Continuous Integration Pipeline](https://docs.microsoft.com/en-us/azure/architecture/example-scenario/apps/devops-dotnet-webapp), but it is a good practice to run them prior opening a [Pull Request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests).
+
+## Limitation of custom tools
+
+If you decide to include [Azure-cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) in your Dev Container, developers will be able to run commands against their tenant. However, to make developers' life easier, we could go further by letting them prefill their connection information, such as the `tenant ID` and the `subscription ID` in a secure and persistent way (do not forget that your Dev Container, being a [Docker](https://www.docker.com/) container, might get deleted, or the image could be rebuilt, hence, all customization *inside* will be lost).
+
+One way to achieve this is to leverage environment variables, with untracked `.env` file part of the solution being injected in the Dev Container.
+
+Consider the following files structure:
+
+```bash
+My Application  # main repo directory
+└───.devcontainer
+|       ├───Dockerfile
+|       ├───docker-compose.yml
+|       ├───devcontainer.json
+└───config
+|       ├───.env
+|       ├───.env-sample
+```
+
+The file `config/.env-sample` is a tracked file where anyone can find environment variables to set (with no values, obviously):
+
+```bash
+TENANT_ID=
+SUBSCRIPTION_ID=
+```
+
+Then, each developer who clones the repository can create the file `config/.env` and fills it in with the appropriate values.
+
+Under the folder `.devcontainer`, we introduce a `docker-compose.yml` file to be able to inject this `.env` file
+
+```yaml
+version: '3'
+services:
+  my-workspace:
+    env_file: ../config/.env
+    build:
+      context: .
+      dockerfile: Dockerfile
+    command: sleep infinity
+```
+
+And in order to use the `docker-compose.yml` file, we need to adjust `devcontainer.json`:
+
+```json
+{
+    "name": "My Application",
+    "dockerComposeFile": ["docker-compose.yml"],
+    "service": "my-workspace"
+    ...
+}
+```
```

Comment:
You are correct, I could use instead:

```yaml
"runArgs": ["--env-file","config/.env"],
```

I kept docker-compose from our previous work because it had more things (dns, port forwarding ...). But for this sample, let's keep it simple. I will update accordingly